lib/jxl/dec_context_map.cc

```diff
@@ -64,14 +64,16 @@ Status DecodeContextMap(std::vector<uint8_t>* context_map, size_t* num_htrees,
                          /*disallow_lz77=*/context_map->size() <= 2));
     ANSSymbolReader reader(&code, input);
     size_t i = 0;
+    uint32_t maxsym = 0;
     while (i < context_map->size()) {
-      uint32_t sym = reader.ReadHybridUint(0, input, dummy_ctx_map);
-      if (sym >= kMaxClusters) {
-        return JXL_FAILURE("Invalid cluster ID");
-      }
+      uint32_t sym = reader.ReadHybridUintClustered<1>(dummy_ctx_map[0], input);
```

Comment:
Can we avoid inlining the context map logic in all the fast-path callers?

Also, please do not use `1` for booleans and please add `/*use_lz77=*/` comments.

Reply:
The logic being that we're doing `map[idx]` ourselves instead of calling it with the arguments `idx, map`?
I guess it could be avoided by defining something like `ReadHybridUintInlined` that is exactly like the unclustered `ReadHybridUint` but will be inlined.

I'll update the style — even though I personally don't really like those `/*param=*/` comments, at least not when there are only a few parameters (when there are 10 then counting positions is more annoying). I don't really like comments in the middle of code in general, but especially not ones that use /* */ because they then make it impossible to use /* */ to temporarily comment out some larger section of code...